### ISSUE TITLE
GetAllTopLevelWindows times out if the process was closed

### ISSUE THREAD

**Issue by aristotelos** [FlaUI/FlaUI#596]
**Describe the bug**
I tried to create some code that closes a window of the application and then gets the list of remaining windows that are open:

```
myWindow.Close();

var remainingWindows = session.App.GetAllTopLevelWindows(session.Automation);
```

However, the `GetAllTopLevelWindows` call in that case times out with an exception after a long time of hanging.

A workaround I applied was to wait for some time after the close to see if the process exits:

```
myWindow.Close();

// Workaround: When closing the last window of the application, the `GetAllTopLevelWindows` function times out with an exception
// Therefore wait for some time
await Wait.Until(() => session.App.HasExited, TimeSpan.FromMilliseconds(2000));

var remainingWindows = session.App.GetAllTopLevelWindows(session.Automation);
```

See https://github.com/FlaUI/FlaUI/blob/7dbe9ae7a9e3d869095c9935804499a593d7d318/src/FlaUI.WebDriver/Controllers/WindowController.cs#L35-L41

**Code snippets**

See above.

**Additional context**

Using UIA3.

**Comment by aristotelos** [comment]
Yes, it's in the NativeElement.FindAllBuildCache call called from UIA3FrameworkAutomationElement.FindAll: 
![image](https://github.com/FlaUI/FlaUI/assets/2814758/93854774-9a6f-46bc-967b-6505714bf90b)

**Comment by sancanella** [comment]
Same here. The instruction blocks the process. Sometimes takes seconds and sometimes minutes. Here is the call stack:
![image](https://github.com/FlaUI/FlaUI/assets/15984018/524edcd9-b16b-4a10-82f6-5406e69f1b96)
